homeassistant/components/risco/binary_sensor.py

```diff
@@ -28,16 +28,13 @@ async def async_setup_entry(
     if is_local(config_entry):
         local_data: LocalData = hass.data[DOMAIN][config_entry.entry_id]
         async_add_entities(
-            RiscoLocalBinarySensor(local_data.system.id, zone_id, zone)
-            for zone_id, zone in local_data.system.zones.items()
-        )
-        async_add_entities(
-            RiscoLocalAlarmedBinarySensor(local_data.system.id, zone_id, zone)
-            for zone_id, zone in local_data.system.zones.items()
-        )
-        async_add_entities(
-            RiscoLocalArmedBinarySensor(local_data.system.id, zone_id, zone)
+            entity
             for zone_id, zone in local_data.system.zones.items()
+            for entity in (
+                RiscoLocalBinarySensor(local_data.system.id, zone_id, zone),
+                RiscoLocalAlarmedBinarySensor(local_data.system.id, zone_id, zone),
+                RiscoLocalArmedBinarySensor(local_data.system.id, zone_id, zone),
+            )
```

Comment:
You could use `itertools.chain` here but there isn't enough of them to matter

Reply:
It might even be a bit simpler to iterate `RiscoLocalBinarySensor`, `RiscoLocalAlarmedBinarySensor`, and `RiscoLocalArmedBinarySensor` since all the arguments they take are the same.

Reply:
what do you mean?

Reply:
I'd probably write it like this https://github.com/home-assistant/core/commit/9dd9f184b5901e3c6b8059beae082917af851791

Reply:
Got it. I feel like it might be a bit harder to read, though. Not by a lot, but still harder.